drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,663 @@
+        DEBUG_PUTS("[nrf24l01p_ng] _init(): spi_init_cs() failed");
+        return -EIO;
+    }
+    if (gpio_init(dev->params.pin_ce, GPIO_OUT) < 0) {
+        DEBUG_PUTS("[nrf24l01p_ng] _init(): gpio_init() failed");
+        return -EIO;
+    }
+    gpio_clear(dev->params.pin_ce);
+    if (nrf24l01p_ng_acquire(dev) < 0) {
+        DEBUG_PUTS("[nrf24l01p_ng] _init(): nrf24l01p_ng_acquire() failed");
+        return -EIO;
+    }
+    if (dev->state != NRF24L01P_NG_STATE_POWER_DOWN) {
+        nrf24l01p_ng_transition_to_power_down(dev);
+    }
+    nrf24l01p_ng_flush_tx(dev);
+    nrf24l01p_ng_flush_rx(dev);
+    uint8_t aw = NRF24L01P_NG_ADDR_WIDTH;
+    uint8_t bc[] = NRF24L01P_NG_BROADCAST_ADDR;
+    luid_get_lb(dev->urxaddr.rxaddrpx.rx_p0, aw);
+     /* "The LSByte must be unique for all six pipes" [datasheet p.38] */
+    if (dev->urxaddr.rxaddrpx.rx_p0[aw - 1] == bc[aw - 1]) {
+        luid_get_lb(dev->urxaddr.rxaddrpx.rx_p0, aw);
+    }
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_RX_ADDR_P0,
+                           dev->urxaddr.rxaddrpx.rx_p0, aw);
+    memcpy(dev->urxaddr.rxaddrpx.rx_p1, bc, aw);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_RX_ADDR_P1,
+                           dev->urxaddr.rxaddrpx.rx_p1, aw);
+    uint8_t setup_aw = NRF24L01P_NG_FLG_AW(nrf24l01p_ng_valtoe_aw(aw));
+    uint8_t rf_setup =
+        NRF24L01P_NG_FLG_RF_DR(dev->params.config.cfg_data_rate) |
+        NRF24L01P_NG_FLG_RF_PWR(dev->params.config.cfg_tx_power);
+    uint8_t setup_retr =
+        NRF24L01P_NG_FLG_ARD(dev->params.config.cfg_retr_delay) |
+        NRF24L01P_NG_FLG_ARC(dev->params.config.cfg_max_retr);
+    uint8_t rf_ch = NRF24L01P_NG_FLG_RF_CH(dev->params.config.cfg_channel);
+    uint8_t en_rxaddr = NRF24L01P_NG_FLG_ERX_P0 | NRF24L01P_NG_FLG_ERX_P1;
+    uint8_t config = NRF24L01P_NG_FLG_CRCO(dev->params.config.cfg_crc);
+    uint8_t features = NRF24L01P_NG_FLG_EN_DYN_ACK |
+                       NRF24L01P_NG_FLG_EN_DPL     |
+                       NRF24L01P_NG_FLG_EN_ACK_PAY;
+    uint8_t en_aa = NRF24L01P_NG_FLG_ENAA_P0 |NRF24L01P_NG_FLG_ENAA_P1;
+    uint8_t endp = NRF24L01P_NG_FLG_DPL_P0 | NRF24L01P_NG_FLG_DPL_P1;
+    uint8_t tx_addr[] = NRF24L01P_NG_DEFAULT_TX_ADDR;
+    uint8_t status = NRF24L01P_NG_FLG_IRQ;
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_SETUP_AW, &setup_aw, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_RF_SETUP, &rf_setup, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_SETUP_RETR, &setup_retr, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_RF_CH, &rf_ch, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_EN_RXADDR, &en_rxaddr, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_CONFIG, &config, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_FEATURES, &features, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_EN_AA, &en_aa, 1);
+    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_DYNPD, &endp, 1);
```

Comment:
Never felt the need for a 

```C
static inline uint8_t nrf24l01p_ng_reg8_write(const nrf24l01p_ng_t *dev, uint8_t reg, uint8_t data) {
    return nrf24l01p_ng_write_reg(dev, reg, &data, 1);
}
```

:wink:

Reply:
Ok